jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java

```diff
@@ -412,10 +442,39 @@ private void selectNextOfType() {
 		}
 	}
 
+	public void setDock(ShortPoint2D requestedDockPosition) {
+		final ISelectable selected = currentSelection.getSingle();
+		if (selected instanceof Building) {
+			DockPosition dockPosition = grid.findDockPosition(requestedDockPosition);
+			if (dockPosition == null) {
+				connector.playSound(116, 1); // this dock position is not at the coast
```

Comment:
Although this is a quick way of doing it, I'd prefer to not have any sound indexes hardcoded in the settlers logic. We have had problems with differing indexes in different versions of the original settlers files, which is very hard to track if the numbers are hardcoded in logic.

The best way would be to send a chat message that the dock  cannot be placed there. Messages (like the attacked message) can then have sounds.